app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,51 @@
+class TasksController < ApplicationController
+  before_action :find_task, only: [:show, :destroy, :edit, :update]
+
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+
+    if @task.save
+      redirect_to task_path(@task) notice: 'Task was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @task.update(task_params)
+      redirect_to @task, notice: 'Task was successfully updated.'
+    else
+      render :new
+  end
+
+  def destroy
+    @task.destroy
+    redirect_to tasks_path
+  end
+
+  private
+
+  def find_task
+    @task = Task.find(params[:id])
+  end
+
+  #to complete when creating form
```

Comment:
don't think this should be commented out, we can edit as needed, it will break the method if commented